core/vtk/ttkMetricDistortion/ttkMetricDistortion.cpp

```diff
@@ -0,0 +1,250 @@
+  }
+  return 0;
+}
+
+/**
+ * Pass VTK data to the base code and convert base code output to VTK
+ *
+ * This method is called during the pipeline execution to update the
+ * already initialized output data objects based on the given input
+ * data objects and filter parameters.
+ *
+ * Note:
+ *     1) The passed input data objects are validated based on the information
+ *        provided by the FillInputPortInformation method.
+ *     2) The output objects are already initialized based on the information
+ *        provided by the FillOutputPortInformation method.
+ */
+int ttkMetricDistortion::RequestData(vtkInformation *ttkNotUsed(request),
+                                     vtkInformationVector **inputVector,
+                                     vtkInformationVector *outputVector) {
+
+  // --------------------------------------------------------------------------
+  // Get input object from input vector
+  // --------------------------------------------------------------------------
+  // Load Surface
+  vtkPolyData *inputSurface = vtkPolyData::GetData(inputVector[0]);
+  if(!inputSurface)
+    return 0;
+  auto noPoints = inputSurface->GetNumberOfPoints();
+  std::vector<std::vector<double>> surfacePoints(
+    noPoints, std::vector<double>(3));
+  for(unsigned int i = 0; i < noPoints; ++i) {
+    double point[3];
+    inputSurface->GetPoints()->GetPoint(i, point);
+    for(unsigned int j = 0; j < 3; ++j)
+      surfacePoints[i][j] = point[j];
+  }
+  auto noCells = inputSurface->GetNumberOfCells();
+  std::vector<std::vector<int>> surfaceCells(noCells);
+  for(unsigned int i = 0; i < noCells; ++i) {
+    auto noCellPoints = inputSurface->GetCell(i)->GetNumberOfPoints();
+    surfaceCells[i] = std::vector<int>(noCellPoints);
+    for(unsigned int j = 0; j < noCellPoints; ++j)
+      surfaceCells[i][j] = inputSurface->GetCell(i)->GetPointId(j);
+  }
+
+  // Detect boundary
+  std::vector<std::vector<int>> edgeBelonging(
+    noPoints, std::vector<int>(noPoints, 0));
+  for(unsigned int i = 0; i < noCells; ++i) {
+    auto noCellPoints = inputSurface->GetCell(i)->GetNumberOfPoints();
+    if(noCellPoints < 3 or noCellPoints > 4)
+      continue;
+    unsigned int noCellEdges = inputSurface->GetCell(i)->GetNumberOfEdges();
+    for(unsigned int j = 0; j < noCellEdges; ++j) {
+      auto p0 = inputSurface->GetCell(i)->GetEdge(j)->GetPointId(0);
+      auto p1 = inputSurface->GetCell(i)->GetEdge(j)->GetPointId(1);
+      ++edgeBelonging[p0][p1];
+      ++edgeBelonging[p1][p0];
+    }
+  }
+  std::vector<bool> isPointBoundary(noPoints, false);
+  for(unsigned int i = 0; i < edgeBelonging.size(); ++i) {
+    for(unsigned int j = 0; j < edgeBelonging[i].size(); ++j) {
+      if(edgeBelonging[i][j] == 1) {
+        isPointBoundary[i] = true;
+        isPointBoundary[j] = true;
+      }
+    }
+  }
+
+  // Load Table
+  vtkTable *distanceMatrixVTK = vtkTable::GetData(inputVector[1]);
+  std::vector<std::vector<double>> distanceMatrix;
```

Comment:
Same here, no need to copy the input data.
Raw pointers can be extracted. Please see other TTK filters using vtkTables as an input.

Reply:
Done in the next commit.